src/hdf5/attribute.cpp

```diff
@@ -0,0 +1,76 @@
+//
+// ICRAR - International Centre for Radio Astronomy Research
+// (c) UWA - The University of Western Australia, 2017
+// Copyright by UWA (in the framework of the ICRAR)
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+/**
+ * @file
+ *
+ * C++ wrapper for HDF5 attributes
+ */
+
+#include "hdf5/attribute.h"
+
+namespace shark {
+namespace hdf5 {
+
+Attribute::Attribute(hid_t handle) : Entity(H5I_ATTR, handle) {}
+
+Attribute::Attribute(const Location& group, const std::string& name) :
+		Attribute(H5Aopen(group.getId(), name.c_str(), H5P_DEFAULT)) {
+}
+
+Attribute::~Attribute() {
+	H5Aclose(getId());
```

Comment:
Related to the assertion of return values, we don't want to throw in a deconstructor, but not sure if it is worth logging a failure here - it probably means we've got in a bad state at some point.

Reply:
I also now check these and log an error if necessary :slightly_smiling_face: